### ISSUE TITLE
Refactor results table

### ISSUE THREAD

**Issue by jzahedieh** [jzahedieh/django-tennis-ladder#17]
Messy, inefficient and downright confusing. this is such core functionality it needs to be easier to understand. Part of the problem I am facing is that I cannot think of an efficient way of achieving of what needs to be done, the process will be documented on this issue to hopefully clarify it and improve it.
### Diagram of current process

![issue-19-table-old-process](https://cloud.githubusercontent.com/assets/2098908/3946055/903a1df8-265e-11e4-9aef-a59ae986df47.jpg)
### Process breakdown

Black: Loops leagues and opponents which are basically the same, a list of Player objects. So for each Player it loops through all their possible opponents to find their results.

Orange: The league players Results are loaded, then all the results are looped to see if they match the current opponent (which is being looped), the result is then filled.

tl;dr loop hell :(
### Current process breakdown

_todo_
### New process breakdown

_todo_

**Comment by jzahedieh** [comment]
Move a lot of the logic/calculations to the database, which reduces query numbers and increases performance
